algorithms/CPlusPlus/Trees/binary-search-tree.cpp

```diff
@@ -158,83 +159,155 @@ bool isperfect(TreeNode* root){
 
         //if(perfect(root))                           { return true; }
         if(perfect_recursive(root, depth(root) - 1)) { return true; }
-	return false;
+    return false;
 }
 
+void leaf_nodes(TreeNode* root) //Print all leafnode in BST
+{ 
+    /* * Node which does not have a child is called as LEAF Node.
+       * Printing the nodes whose left and right pointer are null.
+       * @params: `root` root/parent node of the tree
+    */
+   if (!root) 
+       return; 
+
+   if (root->left==NULL && root->right==NULL) 
+   { 
+       std::cout<<root->data<< " "; 
+       return; 
+   } 
+
+   if (root->left) 
+       leaf_nodes(root->left); 
+
+   if (root->right) 
+       leaf_nodes(root->right); 
+} 
+
+void f_min(TreeNode* root) //Find Minimum element from root
```

Comment:
**Suggestion :**
in are general used case, one would rather use this function to get the minimum value in the tree and use it somewhere else. 

So isn't it better for this function ( and `f_max ()` ) to return the minimum value to the caller and let the caller use it for whatever it want the minimum value for ?

Reply:
Ok sir, I will work on it. I will try to add delete function. There we need minimum value.
Thank you for suggestion

Reply:
It's nice that you also added a used case in the program also.